### PR TITLE
Testing file_gen.go

### PR DESCRIPTION
**Pull request by alicjakwie**

### CODE REVIEW

File: hermes/file_gen_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package hermes
+
+import(
+	//"fmt"
+	"testing"
+	"math/rand"
+	"strconv"
+	"time"
+	//"crypto/sha1"
```
Comment (cianbgoog) [inline hermes/file_gen_test.go]: You can just delete these, rather than hashing them out :-)

File: hermes/file_gen_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	file.contents = global_contents;
+	result := generate_file_checksum(file)
+	if result != result_expected {
+		t.Errorf("generate_file_checksum(\"jhfvjhdfjhfjjhjhdfvjvcvfjh\") failed expected %v got %v", result_expected, result)
+	}
+}
+
+func TestFileName(t *testing.T){
+	rand.Seed(time.Now().UnixNano())
+	var file_ID int = rand.Intn(40)+10;
```
Comment (cianbgoog) [inline hermes/file_gen_test.go]: Don't make your tests random, it'll be a source of continual pain.

If you want randomy style tests for breadth of coverage, look at https://golang.org/pkg/testing/quick/, but if you've doing that, first write a test that uses hard coded values.

**Review by cianbgoog — COMMENTED** [review]
You should combine this with #34

File: hermes/file_gen_test.go (hunk 1 above)
Comment (deniadelcid) [reply hermes/file_gen_test.go]: Yup, if you need them in the future running gofmt should automatically add these for you.

File: hermes/file_gen_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+func TestChecksum(t *testing.T){
+	var global_contents string = "jhfvjhdfjhfjjhjhdfvjvcvfjh"
+	var result_expected string = "_68f3caf439065824dcf75651c202e9f7c28ebf07"
+	file := HermesFile{}
+	file.contents = global_contents;
+	result := generate_file_checksum(file)
+	if result != result_expected {
+		t.Errorf("generate_file_checksum(\"jhfvjhdfjhfjjhjhdfvjvcvfjh\") failed expected %v got %v", result_expected, result)
```
Comment (deniadelcid) [inline hermes/file_gen_test.go]: In case you want to change your file contents in the future, it's best to use the variable. It's also less human-error prone.

Ditto for the errors in TestFileName.
```suggestion
		t.Errorf("generate_file_checksum(\"%s\") failed expected %v got %v", file.contents, want, got)
```

File: hermes/file_gen_test.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	"math/rand"
+	"strconv"
+	"time"
+	//"crypto/sha1"
+)
+
+
+func TestChecksum(t *testing.T){
+	var global_contents string = "jhfvjhdfjhfjjhjhdfvjvcvfjh"
+	var result_expected string = "_68f3caf439065824dcf75651c202e9f7c28ebf07"
```
Comment (deniadelcid) [inline hermes/file_gen_test.go]: - If you are declaring and assigning in the same line, just use the := operator
- For result comparisons, it's more readable to use "want" and "got" for your variable names. To specify what these actually represent you can add an inline comment.

Ditto for the result_expected variable from TestFileName function.
```suggestion
	want := "_68f3caf439065824dcf75651c202e9f7c28ebf07"  // The expected checksum result
```

File: hermes/file_gen_test.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	file := HermesFile{}
+	file.contents = global_contents;
```
Comment (deniadelcid) [inline hermes/file_gen_test.go]: If you're just using global_contents to assign your test contents to file.contents, then consider just skipping the declaration of global_contents and just assigning the string to file.contents directly. 

(P.S. Remember to delete line 10 if/when accepting the suggested change below)

```suggestion
	file.contents =  "jhfvjhdfjhfjjhjhdfvjvcvfjh"
```

File: hermes/file_gen_test.go (hunk 2 above)
Comment (deniadelcid) [inline hermes/file_gen_test.go]: - As mentioned above, use := when declaring and assigning in the same line. Ditto for the assignments below. Alternatively, you can assign them as a block: https://golang.org/doc/effective_go.html#variables.
- You don't need a semi-colon (https://golang.org/doc/effective_go.html#semicolons)

```suggestion
	file_ID := rand.Intn(40)+10
```

File: hermes/file_gen_test.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	result := generate_file_checksum(file)
```
Comment (deniadelcid) [inline hermes/file_gen_test.go]: Similar to the "want" from above, you can rename this one to "got".

Please do the same for the relevant variables in TestFileName.

```suggestion
	got := generate_file_checksum(file)
```

File: hermes/file_gen_test.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
+package hermes
+
+import(
+	"testing"
+	"strconv"
+)
+
```
Comment (evanSpendlove) [inline hermes/file_gen_test.go:20]: Remove whitespace.
```suggestion
```

File: hermes/file_gen_test.go [hunk 8]
```diff
@@ -0,0 +1,38 @@
+package hermes
```
Comment (evanSpendlove) [inline hermes/file_gen_test.go:14]: Add license header, author line and description of file as a comment.

File: hermes/file_gen_test.go [hunk 9]
```diff
@@ -1,39 +1,38 @@
...
-	"time"
-	//"crypto/sha1"
 )
 
 
 func TestChecksum(t *testing.T){
-	var global_contents string = "jhfvjhdfjhfjjhjhdfvjvcvfjh"
-	var result_expected string = "_68f3caf439065824dcf75651c202e9f7c28ebf07"
+	global_contents := "jhfvjhdfjhfjjhjhdfvjvcvfjh"
+	want := "_68f3caf439065824dcf75651c202e9f7c28ebf07" //expected checksum result
```
Comment (deniadelcid) [inline hermes/file_gen_test.go]: Leave 2 spaces before an inline comment.
```suggestion
	want := "_68f3caf439065824dcf75651c202e9f7c28ebf07"  //expected checksum result
```

File: hermes/file_gen_test.go [hunk 10]
```diff
@@ -1,39 +1,38 @@
...
 func TestFileName(t *testing.T){
-	rand.Seed(time.Now().UnixNano())
-	var file_ID int = rand.Intn(40)+10;
-	var fake_checksum string ="_abba"
-	var result_expected string =  "Hermes_"+strconv.Itoa(file_ID)+"_abba"
-	result := generate_file_name(file_ID, fake_checksum)
-	if result != result_expected {
-		t.Errorf("generate_file_name(%v, \"abba\") failed expected %v got %v", file_ID, result_expected, result)
+	file := HermesFile{}
+	file_ID := 23;
```
Comment (deniadelcid) [inline hermes/file_gen_test.go]:
```suggestion
	file_id := 23;
```

Comment (deniadelcid) [inline hermes/file_gen_test.go]: Maybe you want to create a struct (say "TestHermesFile) with a HermesFile, file_id, fake_checksum, and expected_file_names, per se. Then you create a table with several test cases (make sure to test your edge cases too) that then you iterate over.

You can reference this site [1] for some guidance on how to do this: https://gobyexample.com/testing.

**Review by cianbgoog — APPROVED** [review]

File: hermes/file_gen_test.go [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+	checksum string
+	want string
+}
+
+var test_unit_table [5]FileNameTestUnit
+
+// Function completes the table of HermesFileTestUnit by iterating over its entries
+func completeTestUnitTable(){
+	for i:=0; i<5; i++ {
+		test_unit_table[i].id = i*10+1
```
Comment (cianbgoog) [inline hermes/file_gen_test.go]: It's probably better, and easier, to just put these in line, rather than using a loop to do it.

**Review by cianbgoog — APPROVED** [review]
